state/watcher.go

```diff
@@ -2733,3 +2724,51 @@ func (w *offeredApplicationsWatcher) loop() (err error) {
 func (w *offeredApplicationsWatcher) Changes() <-chan []string {
 	return w.out
 }
+
+// WatchRemoteRelations returns a StringsWatcher that notifies of changes to
+// the lifecycles of the remote relations in the model.
+func (st *State) WatchRemoteRelations() StringsWatcher {
+	filter := func(id interface{}) bool {
+		id, err := st.strictLocalID(id.(string))
+		if err != nil {
+			return false
+		}
+
+		type relationDoc struct {
+			Endpoints []Endpoint
+		}
+		endpointFields := bson.D{{"endpoints", 1}}
+		var doc relationDoc
+
+		relations, closer := st.getCollection(relationsC)
+		defer closer()
+		if err := relations.FindId(id).Select(endpointFields).One(&doc); err != nil {
+			if err == mgo.ErrNotFound {
+				// If the relation can't be found, we allow it to be processed by the watcher.
+				// If it was a remote relation, it will trigger an event on transition to dead.
+				// If it wasn't a remote relation, it will result in a no-op as the relation was
```

Comment:
this feels a bit kludgey. can we keep track of which relations we've told the client about already (i.e. as a map/set closed over by this function)?